### PR TITLE
mc: consistent exit status across commands.

### LINKED ISSUE #1880: `mc` returns inconsistent exit status on same kind of error
## Expected behaviour

mc should exit with non-zero exit code for "connection refused" errors (i.e. when the server is down).

## Actual behaviour

In some cases mc exits with 0 exit status when server is down. This makes it difficult to use in scripts.

## Steps to reproduce the behaviour

```
aditya@donatello-laptop:~/Code/minio-code/data$ echo "tiny object" | mc pipe dminio1/testbucket/one
mc: <ERROR> Unable to write to one or more targets. Get http://127.0.0.1:9001/testbucket/?location=: dial tcp 127.0.0.1:9001: getsockopt: connection refused
aditya@donatello-laptop:~/Code/minio-code/data$ echo $?
1
aditya@donatello-laptop:~/Code/minio-code/data$ mc ls dminio1
mc: <ERROR> Unable to list folder. Get http://127.0.0.1:9001/: dial tcp 127.0.0.1:9001: getsockopt: connection refused
aditya@donatello-laptop:~/Code/minio-code/data$ echo $?
0

```

## mc version
```
$ mc version
Version: 2016-11-11T19:01:44Z
Release-tag: DEVELOPMENT.2016-11-11T19-01-44Z
Commit-id: 2895b0accaad4465128e4b950bf048494de36a9f
```
## System information
Linux.

### PR DESCRIPTION
**Pull request by harshavardhana**
Fixes #1880

### CODE REVIEW

File: cmd/error.go [hunk 1]
```diff
@@ -76,11 +77,20 @@ func fatalIf(err *probe.Error, msg string) {
 	console.Fatalln(fmt.Sprintf("%s %s", msg, err))
 }
 
+func exitIf(i int) {
+	if errorIfExitStatus {
+		os.Exit(i)
+	}
```
Comment (vadmeste) [inline cmd/error.go]: Does it make sense to add os.Exit(0) here for a better code clarity ?

Comment (harshavardhana) [inline cmd/error.go]: Thinking..

File: cmd/error.go [hunk 2]
```diff
@@ -76,11 +77,22 @@ func fatalIf(err *probe.Error, msg string) {
 	console.Fatalln(fmt.Sprintf("%s %s", msg, err))
 }
 
+var errorIfExitStatus int
```
Comment (abperiasamy) [inline cmd/error.go]: It is a poor programming practice to declare global variables like this one. Global variables that are accessed across files are to be placed in globals.go. That way we know the ones that we have are must have.

File: cmd/error.go [hunk 3]
```diff
@@ -76,11 +77,22 @@ func fatalIf(err *probe.Error, msg string) {
...
+
+func exitIf(exitStatus int) {
```
Comment (abperiasamy) [inline cmd/error.go]: Besides not having comments, Why do we need a wrapper around os.Exit?

File: cmd/mb-main.go [hunk 4]
```diff
@@ -117,19 +117,18 @@ func mainMakeBucket(ctx *cli.Context) {
...
-		err = clnt.MakeBucket(region)
-		if err != nil {
+		if err == nil {
+			// Make bucket.
+			err = clnt.MakeBucket(region)
+			if err == nil {
+				// Successfully created a bucket.
+				printMsg(makeBucketMessage{Status: "success", Bucket: targetURL})
+				continue
+			}
```
Comment (abperiasamy) [inline cmd/mb-main.go]: This is also a poor coding style compared to the original code.

File: cmd/error.go [hunk 5]
```diff
@@ -76,11 +77,22 @@ func fatalIf(err *probe.Error, msg string) {
...
+}
+
 // errorIf synonymous with fatalIf but doesn't exit on error != nil
 func errorIf(err *probe.Error, msg string) {
 	if err == nil {
 		return
 	}
+
+	// Set exit status.
+	errorIfExitStatus = 1
```
Comment (abperiasamy) [inline cmd/error.go]: Avoid global variables like this one and use functional style coding.

File: cmd/mb-main.go (hunk 4 above)
Comment (harshavardhana) [inline cmd/mb-main.go]: Done

**Comment by harshavardhana** [comment]
Closing the PR, will be fixing this properly with CLI changes.

**Pull request closed without merging**